go/vt/vtctld/api.go

```diff
@@ -60,8 +61,16 @@ const (
 	jsonContentType = "application/json; charset=utf-8"
 )
 
-// TabletWithURL wraps topo.Tablet and adds a URL property.
-type TabletWithURL struct {
+// TabletStats represents realtime stats from a discovery.LegacyTabletStats struct.
+type TabletStats struct {
+	LastError string                 `json:"last_error,omitempty"`
+	Realtime  *querypb.RealtimeStats `json:"realtime,omitempty"`
+	Serving   bool                   `json:"serving"`
+	Up        bool                   `json:"up"`
+}
+
+// TabletWithStatsAndURL wraps topo.Tablet, adding a URL property and optional realtime stats.
+type TabletWithStatsAndURL struct {
```

Comment:
Similar comment about exporting the struct as with `TabletStats`.

Reply:
@deepthi see earlier comment. I just noticed `NewTabletWithStatsAndURL` doesn't need to be public, I'll make this a private function 👍

Reply:
> I just noticed `NewTabletWithStatsAndURL` doesn't need to be public, I'll make this a private function 👍

https://github.com/vitessio/vitess/pull/6569/commits/1c653041aaa3e28d7dba8c94da8a3bda77ba6eeb makes `NewTabletWithStatsAndURL` a private func